src/vivarium/framework/results/stratification.py

```diff
@@ -71,13 +71,19 @@ def __call__(self, population: pd.DataFrame) -> pd.DataFrame:
             mapped_column = self.mapper(population[self.sources])
         else:
             mapped_column = population[self.sources].apply(self.mapper, axis=1)
-
-        if mapped_column.isnull().any():
-            raise ValueError(f"Null values mapped to {self.name}.")
         unknown_categories = set(mapped_column) - set(
             self.categories + self.excluded_categories
         )
         if unknown_categories:
+            # Reduce all nans to a single one
+            single_nan_list = (
+                [mapped_column[mapped_column.isna()].iat[0]]
+                if mapped_column.isna().any()
+                else []
+            )
+            unknown_categories = single_nan_list + [
+                cat for cat in unknown_categories if not pd.isna(cat)
+            ]
```

Comment:
Why can't this be:
```python
unknown_categories = [
    cat for cat in unknown_categories if not pd.isna(cat)
]
if mapped_column.isna().any():
    unknown_categories.append(np.nan)
```

Reply:
I feel like pd.isna wasn't working as expected but now I'm not so sure. I'll try again

Reply:
Ok, yeah that works and is way better. I did stick w/ appending the type of NaN since that could potentially shed light on the issue but same basic logic